src/App.js

```diff
@@ -1,42 +1,76 @@
 //imports
-import React from 'react';
+import React, { Component } from 'react';
 import Header from './Header/Header.js';
 import FilmCollection from './FilmCollection/FilmCollection';
 import Footer from './Footer/Footer.js';
 import ResultsArea from './ResultsArea/ResultsArea';
 import ErrorBoundary from './ErrorBoundary/ErrorBoundary';
+import { fetchRequestIfNeeded } from './redux-utils/asyncActionUtils';
 import { connect } from 'react-redux';
+import { Route, Switch } from 'react-router-dom';
 import './style.css';
 
-const App = (props) => {
-
-    const { films, isFilmClicked, genre, filmsSameGenre } = props;
-    const filmsNumber = props.films.length;
-
-    return (
-        <div className='app'>
-            <ErrorBoundary>
-                <Header isFilmClicked={isFilmClicked} />
-            </ErrorBoundary>
-            <ErrorBoundary>
-                <ResultsArea isFilmClicked={isFilmClicked} genre={genre} filmsNumber={filmsNumber} />
-            </ErrorBoundary>
-            <ErrorBoundary>
-                <FilmCollection filmsNumber={filmsNumber} films={films} filmsSameGenre={filmsSameGenre} isFilmClicked={isFilmClicked} />
-            </ErrorBoundary>
-            <Footer/>
-        </div>
-    );
-};
+class App extends Component {
+
+    constructor(props) {
+        super(props);
+        this.props = props;
+    }
+
+    componentDidMount() {
+        const id = this.props.match.params.id;
+        if (id) {
+            this.props.handleUrlChange(id);
+        }
+    }
+
+    render() {
+
+        const { films, genre, filmsSameGenre } = this.props;
+        const filmsNumber = this.props.films.length;
+
+        return (
+            <div className='app'>
+                <ErrorBoundary>
+                    <Switch>
+                        <Route path='/' render={() => <Header /> } />
+                        <Route path='/film/:id' render={() => <Header /> } />
+                        <Route path='/search/Search :value' render={() => <Header /> } />
```

Comment:
What is the reason to use `Route` for header component? You render the same components for each route... It would make a sense if you rendered different components for each route or passed different props to `Header` component.